### PR TITLE
MM-14786 Properly implement Enable NPS Survey setting

### PR DESCRIPTION
**Pull request by hmhealey**
I managed to forget to implement a key part of the plugin settings somehow. This fixes the name of the EnableSurvey setting and ensures that survey-related functions are only ran when that setting is true.

### CODE REVIEW

File: server/configuration.go [hunk 1]
```diff
@@ -80,5 +82,12 @@ func (p *Plugin) OnConfigurationChange() error {
 
 	p.setConfiguration(configuration)
 
+	if p.isActivated() {
```
Comment (hanzei) [inline server/configuration.go]: Why do you need to check this?
1. If the plugin get disabled, the hook will not be called.
2. `p.activated` gets set to `true` at startup.

File: server/survey.go [hunk 2]
```diff
@@ -40,6 +40,16 @@ type adminNotice struct {
 // Note that this only sends an email to admins to notify them that a survey has been scheduled. The web app plugin is
 // in charge of checking and actually triggering the survey.
 func (p *Plugin) checkForNextSurvey(currentVersion semver.Version) bool {
+	p.surveyLock.Lock()
+	defer p.surveyLock.Unlock()
+
+	if !p.getConfiguration().EnableSurvey {
+		// Surveys are disabled, so return false without updating the stored version. If surveys are re-enabled, the
```
Comment (hanzei) [inline server/survey.go]: Great comment! :+1:

File: server/plugin.go [hunk 3]
```diff
@@ -20,18 +20,19 @@ const (
 type Plugin struct {
 	plugin.MattermostPlugin
 
-	// configurationLock synchronizes access to the configuration.
```
Comment (hanzei) [inline server/plugin.go]: Why did you remove this?

File: server/plugin.go [hunk 4]
```diff
@@ -20,18 +20,19 @@ const (
...
 	configurationLock sync.RWMutex
+	configuration     *configuration
 
-	// configuration is the active plugin configuration. Consult getConfiguration and
```
Comment (hanzei) [inline server/plugin.go]: Same question as above?

File: server/plugin.go [hunk 5]
```diff
@@ -20,18 +20,19 @@ const (
...
-	// setConfiguration for usage.
-	configuration *configuration
+	activatedLock sync.RWMutex
+	activated     bool
+
+	surveyLock sync.Mutex
```
Comment (hanzei) [inline server/plugin.go]: Could you please add a comment why this is needed?

File: server/configuration.go (hunk 1 above)
Comment (hmhealey) [reply server/configuration.go]: OnConfigurationChange actually gets called before OnActivate. I'm not sure if that's a bug or intended so that the plugin use its configuration in OnActivate (I also filed [a ticket for that](https://mattermost.atlassian.net/browse/MM-14788). I just added this check to be extra sure since I was worried about any potential weirdness there.

File: server/plugin.go (hunk 3 above)
Comment (hmhealey) [reply server/plugin.go]: I wanted to group the fields to make it easier to see that they were related. Removing the comments probably wasn't the best idea though, so I'll add them back in

File: server/configuration.go (hunk 1 above)
Comment (hanzei) [reply server/configuration.go]: Hm, I see. This feels like something the plugin system should support by default. But your solution is correct.

File: server/plugin.go (hunk 3 above)
Comment (hanzei) [reply server/plugin.go]: You can still group, but keep the comments :wink:.

File: server/configuration.go (hunk 1 above)
Comment (hmhealey) [reply server/configuration.go]: Yeah, I can see reasons for doing it both ways. On one hand, I want to know if the configuration changes to set the plugin's config, even before OnActivate has been called. On the other hand, I assumed that it was only called after OnActivate, and I can see others making the same mistake
